zuul/reporter/github.py

```diff
@@ -75,11 +75,15 @@ def setPullStatus(self, pipeline, item):
         sha = item.change.patchset
         context = pipeline.name
         state = self._github_status_value
+
         url = ''
-        if self.sched.config.has_option('zuul', 'status_url'):
+        if self._status_url:
```

Comment:
This could use some code commentary around the logic used. It's somewhat difficult to follow.